aten/src/ATen/native/cuda/SummaryOps.cu

```diff
@@ -283,11 +295,34 @@ Tensor _bincount_cuda(
   return AT_DISPATCH_INTEGRAL_TYPES(self.type(), "bincount", [&] {
     const auto scalar = weights.type().scalarType();
     if (scalar == ScalarType::Undefined || scalar == ScalarType::Float)
-      return _bincount_cuda_template<scalar_t, float>(self, weights, minlength);
-    return _bincount_cuda_template<scalar_t, double>(
+      return _bincount_cuda_template<scalar_t, float, scalar_t>(self, weights, minlength);
+    return _bincount_cuda_template<scalar_t, double, scalar_t>(
         self, weights.toType(CUDA(kDouble)), minlength);
   });
 }
 
+template <typename scalar_t>
+Tensor _histc_cuda(
```

Comment:
This shouldn't be templatized with scalar_t: You should do a dispatch on the type of the input tensor, as so: https://github.com/pytorch/pytorch/blob/c5cc1e3ab28380d1e1a4a1a0db8a295bd45c0bd8/aten/src/ATen/native/cuda/SummaryOps.cu#L283-L290